src/main/java/org/apache/maven/shared/verifier/Verifier.java

```diff
@@ -115,87 +115,195 @@
+     * use {@link #Verifier(String basedir)},
+     *{@link #setSettingsFile(String settingsFile)} to set settings file,
+     *{@link #setForkJvm(Boolean)} to set forkJvm status and
+     * use {@link #setDefaultCliArguments(String[] defaultCliArguments)} to set settings file
+     *
+     * */
+    @Deprecated
     public Verifier( String basedir, String settingsFile, boolean debug, boolean forkJvm, String[] defaultCliArguments )
         throws VerificationException
     {
-        this( basedir, settingsFile, debug, forkJvm, defaultCliArguments, null );
+        this( basedir );
+        setSettingsFile( basedir );
+        setDebug( debug );
+        setForkJvm( forkJvm );
+        setDefaultCliArguments( defaultCliArguments );
+        setMavenHome( null );
     }
 
+    /**
+     *
+     * @deprecated to be removed
+     * use {@link #Verifier(String basedir)},
+     * {@link #setSettingsFile(String settingsFile)} to set settings file
+     * and {@link #setMavenHome(String mavenHome)}  to set maven home
+     *
+     * */
+    @Deprecated
     public Verifier( String basedir, String settingsFile, boolean debug, String mavenHome )
             throws VerificationException
     {
-        this( basedir, settingsFile, debug, null, DEFAULT_CLI_ARGUMENTS, mavenHome );
+        this( basedir );
+        setForkJvm( null );
+        setDebug( debug );
+        setSettingsFile( settingsFile );
+        setMavenHome( mavenHome );
     }
 
+    /**
+     *
+     * @deprecated to be removed
+     * use {@link #Verifier(String basedir)}
+     * use {@link #setSettingsFile(String settingsFile)} to set settings file
+     * use {@link #setMavenHome(String mavenHome)}  to set maven home
+     * use {@link #setDefaultCliArguments(String[] defaultCliArguments)} to set settings file
+     *
+     * */
+    @Deprecated
     public Verifier( String basedir, String settingsFile, boolean debug, String mavenHome,
                      String[] defaultCliArguments )
         throws VerificationException
     {
-        this( basedir, settingsFile, debug, null, defaultCliArguments, mavenHome );
+        this( basedir );
+        setSettingsFile( settingsFile );
+        setDebug( debug );
+        setForkJvm( null );
+        setDefaultCliArguments( defaultCliArguments );
+        setMavenHome( mavenHome );
+
     }
 
+    /**
+     *
+     * @deprecated to be removed
+     * use {@link #Verifier(String basedir)}
+     * use {@link #setSettingsFile(String settingsFile)} to set settings file
+     * use {@link #setForkJvm(Boolean)}  to set forkJvm status
+     * use {@link #setDefaultCliArguments(String[] defaultCliArguments)} to set CLI Arguments
+     *  use {@link #setMavenHome(String mavenHome)}  to set maven home
+     *
+     * */
+    @Deprecated
     private Verifier( String basedir, String settingsFile, boolean debug, Boolean forkJvm, String[] defaultCliArguments,
             String mavenHome ) throws VerificationException
     {
```

Comment:
Is it still used?
If no we can remove private constructor at all.

Reply:
It is no longer in use, I will remove this.